### PR TITLE
Bug fixes to DSMI loader

### PR DESCRIPTION
**Pull request by neumatho**
Bug fixes to DSMI loader, which fixes these bugs:

- Did always set the calculated panning on the first channel, so it will contain the last calculated value while the rest will have default values.
- For version 11 modules, the loader skip the panning values as it think it's channel remappings. Version 11 also only have 16 panning values while version 12 and higher have 32.
- Could not load version 12 modules, since it did not load the panning values (not tested, since I do not have any modules in this version, but according to Ottos code, version 12 should have 32 bytes of panning values, but libxmp only loaded 16 bytes as channel remaps).
- Added extra unit test for version 13.

### CODE REVIEW

**Comment by AliceLR** [comment]
This looks reasonable at a glance, I'll verify tomorrow.

**Comment by sezero** [comment]
The new test cases are never run, because the following change is missing:

```diff
diff --git a/test-dev/Makefile.in b/test-dev/Makefile.in
index c5aa0f0..d6e933a 100644
--- a/test-dev/Makefile.in
+++ b/test-dev/Makefile.in
@@ -293,3 +293,3 @@
 		  digi \
-		  dsmi dsmi_08 dsmi_09 dsmi_10 dsmi_10_trunc \
+		  dsmi dsmi_08 dsmi_09 dsmi_10 dsmi_10_trunc dsmi_11 dsmi_13 \
 		  dsmi_track0 dsmi_note7f dsmi_pan dsmi_vol \
```

**Comment by neumatho** [comment]
Strange, they ran on my Windows machine. Anyway, I have added them to the makefile.

File: test-dev/data/m/Revelution.amf
Comment (AliceLR) [inline test-dev/data/m/Revelution.amf:1]: ["beat it up" by Sinbad](https://ftp.modland.com/pub/modules/Digital%20Sound%20And%20Music%20Interface/Sinbad/beat%20it%20up.amf) is 25% of the size and is also a DSMI 1.1 that uses hard panning.

Comment (sezero) [reply test-dev/data/m/Revelution.amf:1]: Replace the test file and then merge?

Comment (AliceLR) [reply test-dev/data/m/Revelution.amf:1]: Everything else looks good, yeah. These test files start to add up though, and once they're in the main branch, every `git clone` forever will download them...

**Comment by neumatho** [comment]
Replaced test module

**Review by sezero — APPROVED** [review]

**Comment by sezero** [comment]
OK, this is in now. Thank you.

**Pull request merged**